Description: theme_text_color replaced by theme_fg_color
 there is no theme_text_color in gtk 3.14, this results in dark-on-dark
 unreadable widgets using Adwaita dark variant.
Forwarded: https://bugzilla.gnome.org/show_bug.cgi?id=737340
Author: Jérémy Lal <kapouer@melix.org>
Last-Update: 2014-09-25

--- evolution-3.12.6.orig/addressbook/gui/widgets/e-minicard-label.c
+++ evolution-3.12.6/addressbook/gui/widgets/e-minicard-label.c
@@ -422,7 +422,7 @@ set_colors (EMinicardLabel *label)
 
 		widget = GTK_WIDGET (GNOME_CANVAS_ITEM (label)->canvas);
 
-		e_utils_get_theme_color_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
+		e_utils_get_theme_color_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
 
 		if (label->has_focus) {
 			GdkColor outline, fill;
--- evolution-3.12.6.orig/addressbook/gui/widgets/e-minicard.c
+++ evolution-3.12.6/addressbook/gui/widgets/e-minicard.c
@@ -283,7 +283,7 @@ set_selected (EMinicard *minicard,
 			NULL);
 	} else {
 		e_utils_get_theme_color_color (widget, "theme_bg_color", E_UTILS_DEFAULT_THEME_BG_COLOR, &header);
-		e_utils_get_theme_color_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
+		e_utils_get_theme_color_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
 
 		gnome_canvas_item_set (
 			minicard->rect,
--- evolution-3.12.6.orig/calendar/gui/e-day-view-time-item.c
+++ evolution-3.12.6/calendar/gui/e-day-view-time-item.c
@@ -305,7 +305,7 @@ edvti_draw_zone (GnomeCanvasItem *canvas
 		context, day_view->large_font_desc,
 		pango_context_get_language (context));
 
-	e_utils_get_theme_color (widget, "theme_fg_color,theme_text_color", E_UTILS_DEFAULT_THEME_FG_COLOR, &fg);
+	e_utils_get_theme_color (widget, "theme_fg_color,theme_fg_color", E_UTILS_DEFAULT_THEME_FG_COLOR, &fg);
 	e_utils_get_theme_color (widget, "theme_base_color", E_UTILS_DEFAULT_THEME_BASE_COLOR, &dark);
 
 	/* The start and end of the long horizontal line between hours. */
--- evolution-3.12.6.orig/calendar/gui/e-memo-table.c
+++ evolution-3.12.6/calendar/gui/e-memo-table.c
@@ -502,7 +502,7 @@ memo_table_query_tooltip (GtkWidget *wid
 	e_utils_get_theme_color (widget, "theme_selected_bg_color", E_UTILS_DEFAULT_THEME_SELECTED_BG_COLOR, &sel_bg);
 	e_utils_get_theme_color (widget, "theme_selected_fg_color", E_UTILS_DEFAULT_THEME_SELECTED_FG_COLOR, &sel_fg);
 	e_utils_get_theme_color (widget, "theme_bg_color", E_UTILS_DEFAULT_THEME_BG_COLOR, &norm_bg);
-	e_utils_get_theme_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &norm_text);
+	e_utils_get_theme_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &norm_text);
 
 	box = gtk_box_new (GTK_ORIENTATION_VERTICAL, 0);
 
--- evolution-3.12.6.orig/calendar/gui/e-task-table.c
+++ evolution-3.12.6/calendar/gui/e-task-table.c
@@ -804,7 +804,7 @@ task_table_query_tooltip (GtkWidget *wid
 	e_utils_get_theme_color (widget, "theme_selected_bg_color", E_UTILS_DEFAULT_THEME_SELECTED_BG_COLOR, &sel_bg);
 	e_utils_get_theme_color (widget, "theme_selected_fg_color", E_UTILS_DEFAULT_THEME_SELECTED_FG_COLOR, &sel_fg);
 	e_utils_get_theme_color (widget, "theme_bg_color", E_UTILS_DEFAULT_THEME_BG_COLOR, &norm_bg);
-	e_utils_get_theme_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &norm_text);
+	e_utils_get_theme_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &norm_text);
 
 	box = gtk_box_new (GTK_ORIENTATION_VERTICAL, 0);
 
--- evolution-3.12.6.orig/calendar/gui/e-week-view.c
+++ evolution-3.12.6/calendar/gui/e-week-view.c
@@ -1737,7 +1737,7 @@ e_week_view_set_colors (EWeekView *week_
 
 	e_utils_get_theme_color (widget, "theme_base_color", E_UTILS_DEFAULT_THEME_BASE_COLOR, &base_bg);
 	e_utils_get_theme_color (widget, "theme_bg_color", E_UTILS_DEFAULT_THEME_BG_COLOR, &bg_bg);
-	e_utils_get_theme_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text_fg);
+	e_utils_get_theme_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text_fg);
 	e_utils_get_theme_color (widget, "theme_selected_bg_color", E_UTILS_DEFAULT_THEME_SELECTED_BG_COLOR, &selected_bg);
 	e_utils_get_theme_color (widget, "theme_selected_fg_color", E_UTILS_DEFAULT_THEME_SELECTED_FG_COLOR, &selected_fg);
 	e_utils_get_theme_color (widget, "theme_unfocused_selected_bg_color,theme_selected_bg_color", E_UTILS_DEFAULT_THEME_UNFOCUSED_SELECTED_BG_COLOR, &unfocused_selected_bg);
--- evolution-3.12.6.orig/e-util/e-calendar-item.c
+++ evolution-3.12.6/e-util/e-calendar-item.c
@@ -1436,7 +1436,7 @@ e_calendar_item_draw_month (ECalendarIte
 	pango_layout_set_font_description (layout, font_desc);
 	if (gtk_widget_get_direction (widget) == GTK_TEXT_DIR_RTL)
 		text_x += (7 - 1) * calitem->cell_width;
-	e_utils_get_theme_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &rgba);
+	e_utils_get_theme_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &rgba);
 	gdk_cairo_set_source_rgba (cr, &rgba);
 	for (day = 0; day < 7; day++) {
 		cairo_save (cr);
@@ -1646,7 +1646,7 @@ e_calendar_item_draw_day_numbers (ECalen
 				get_digit_fomat (), digit);
 
 			cairo_save (cr);
-			e_utils_get_theme_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &rgba);
+			e_utils_get_theme_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &rgba);
 			gdk_cairo_set_source_rgba (cr, &rgba);
 			pango_layout_set_font_description (layout, font_desc);
 			pango_layout_set_text (layout, buffer, num_chars);
--- evolution-3.12.6.orig/e-util/e-cell-text.c
+++ evolution-3.12.6/e-util/e-cell-text.c
@@ -725,7 +725,7 @@ ect_draw (ECellView *ecell_view,
 	} else {
 		gboolean color_overwritten = FALSE;
 
-		e_utils_get_theme_color (canvas, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &fg_rgba);
+		e_utils_get_theme_color (canvas, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &fg_rgba);
 		gdk_cairo_set_source_rgba (cr, &fg_rgba);
 
 		if (ect->color_column != -1) {
@@ -820,7 +820,7 @@ ect_draw (ECellView *ecell_view,
 				}
 			} else {
 				e_utils_get_theme_color (canvas, "theme_base_color", E_UTILS_DEFAULT_THEME_BASE_COLOR, &bg_rgba);
-				e_utils_get_theme_color (canvas, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &fg_rgba);
+				e_utils_get_theme_color (canvas, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &fg_rgba);
 			}
 
 			indices[0] = MIN (edit->selection_start, edit->selection_end);
--- evolution-3.12.6.orig/e-util/e-table-click-to-add.c
+++ evolution-3.12.6/e-util/e-table-click-to-add.c
@@ -85,7 +85,7 @@ etcta_style_updated (ETableClickToAdd *e
 
 	e_utils_get_theme_color_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_FG_COLOR, &fg);
 	e_utils_get_theme_color_color (widget, "theme_bg_color", E_UTILS_DEFAULT_THEME_BG_COLOR, &bg);
-	e_utils_get_theme_color_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
+	e_utils_get_theme_color_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
 
 	if (etcta->rect)
 		gnome_canvas_item_set (
@@ -266,7 +266,7 @@ create_rect_and_text (ETableClickToAdd *
 
 	e_utils_get_theme_color_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_FG_COLOR, &fg);
 	e_utils_get_theme_color_color (widget, "theme_bg_color", E_UTILS_DEFAULT_THEME_BG_COLOR, &bg);
-	e_utils_get_theme_color_color (widget, "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
+	e_utils_get_theme_color_color (widget, "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &text);
 
 	if (!etcta->rect)
 		etcta->rect = gnome_canvas_item_new (
--- evolution-3.12.6.orig/e-util/e-text.c
+++ evolution-3.12.6/e-util/e-text.c
@@ -1308,7 +1308,7 @@ e_text_draw (GnomeCanvasItem *item,
 			gdk_cairo_set_source_rgba (cr, &rgba);
 			cairo_paint (cr);
 
-			e_utils_get_theme_color (widget, backdrop ? "theme_unfocused_text_color,theme_text_color" : "theme_text_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &rgba);
+			e_utils_get_theme_color (widget, backdrop ? "theme_unfocused_text_color,theme_fg_color" : "theme_fg_color", E_UTILS_DEFAULT_THEME_TEXT_COLOR, &rgba);
 			gdk_cairo_set_source_rgba (cr, &rgba);
 			cairo_move_to (cr, xpos, ypos);
 			pango_cairo_show_layout (cr, text->layout);
